pkg/detectors/sonarcloud/sonarcloud.go

```diff
@@ -0,0 +1,72 @@
+package sonarcloud
+
+import (
+	"context"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	//Make sure that your group is surrounded in boundry characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"sonarcloud"}) + `\b([0-9a-z]{40})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"sonarcloud"}
+}
+
+// FromData will find and optionally verify SonarCloud secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+
+		s1 := detectors.Result{
+			DetectorType: detectorspb.DetectorType_SonarCloud,
+			Raw:          []byte(resMatch),
+		}
+
+		if verify {
+			req, err := http.NewRequestWithContext(ctx, "GET", "https://"+resMatch+"@sonarcloud.io/api/users/search", nil)
```

Comment:
It looks like there's a validation endpoint: https://sonarcloud.io/web_api/api/authentication?deprecated=false

Reply:
Is the credential that we are testing with as-is from sonarcloud? I'm asking because most have to be base64 encoded before they can be used with basic auth.

Reply:
Yes, the credential is generated in sonarcloud. It is located under My Account->Security. I just updated the endpoint and added organization as parameter since it seems validation endpoint still returns true even without token or credentials.
![image](https://user-images.githubusercontent.com/51393035/162549418-f6c778e0-f121-43fa-bd9a-c8a42321d0c8.png)